### PR TITLE
Setting the endpoint_id of all the uninitialized endpoints to invalid instead of 0

### PR DESCRIPTION
**Pull request by chiragatal**
Setting the endpoint_id of all the uninitialized endpoints to invalid instead of 0

#### Problem
Fixing corner cases where dynamic endpoints are added and removed.

#### Change overview
* Setting the endpoint IDs for uninitialized or removed endpoints to EMBER_AF_INVALID_ENDPOINT instead of setting it to 0.
* reference: https://github.com/project-chip/connectedhomeip/pull/12934#discussion_r769262073

#### Testing
* Added and removed dynamic endpoints multiple times.
* Removed an endpoint and added another one on the same index.

### CODE REVIEW

**Review by dhrishi — APPROVED** [review]

**Comment by dhrishi** [comment]
@bzbarsky-apple PTAL, wrt your comment https://github.com/project-chip/connectedhomeip/pull/12934#discussion_r769262073

**Review by andy31415 — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
So first of all, EMBER_AF_INVALID_ENDPOINT is actually set to a value which is a valid endpoint id (in Matter, which has a different range of endpoint ids than Zigbee).  So that would need to change at the very least.

But past that, I am trying to understand the underlying problem here that #12934 was trying to solve.  Why would endpoint 0 be a dynamic endpoint?  Is this a device with _only_ dynamic endpoints configured?

**Comment by chiragatal** [comment]
Updated the EMBER_AF_INVALID_ENDPOINT to 0xFFFF.

> Is this a device with only dynamic endpoints configured?

Yes, experimenting to create all the endpoints dynamically.

File: src/app/util/attribute-storage.cpp [hunk 1]
```diff
@@ -149,6 +149,10 @@ void emberAfEndpointConfigure(void)
         static_assert(EMBER_AF_ENDPOINT_DISABLED == 0, "We are creating enabled dynamic endpoints!");
         memset(&emAfEndpoints[FIXED_ENDPOINT_COUNT], 0,
                sizeof(EmberAfDefinedEndpoint) * (MAX_ENDPOINT_COUNT - FIXED_ENDPOINT_COUNT));
+        for (ep = FIXED_ENDPOINT_COUNT; ep < MAX_ENDPOINT_COUNT; ep++)
+        {
+            emAfEndpoints[ep].endpoint = EMBER_AF_INVALID_ENDPOINT;
```
Comment (bzbarsky-apple) [inline src/app/util/attribute-storage.cpp:154]:
```suggestion
            emAfEndpoints[ep].endpoint = kInvalidEndpointId;
```

File: src/app/util/attribute-storage.cpp [hunk 2]
```diff
@@ -180,11 +184,15 @@ EmberAfStatus emberAfSetDynamicEndpoint(uint16_t index, EndpointId id, EmberAfEn
     {
         return EMBER_ZCL_STATUS_INSUFFICIENT_SPACE;
     }
+    if (id == EMBER_AF_INVALID_ENDPOINT)
```
Comment (bzbarsky-apple) [inline src/app/util/attribute-storage.cpp:187]:
```suggestion
    if (id == kInvalidEndpointId)
```

File: src/app/util/attribute-storage.cpp [hunk 3]
```diff
@@ -213,15 +221,13 @@ EndpointId emberAfClearDynamicEndpoint(uint16_t index)
 
     index = static_cast<uint8_t>(index + FIXED_ENDPOINT_COUNT);
 
-    if ((index < MAX_ENDPOINT_COUNT) && (emAfEndpoints[index].endpoint != 0) && (emberAfEndpointIndexIsEnabled(index)))
+    if ((index < MAX_ENDPOINT_COUNT) && (emAfEndpoints[index].endpoint != EMBER_AF_INVALID_ENDPOINT) &&
```
Comment (bzbarsky-apple) [inline src/app/util/attribute-storage.cpp:224]:
```suggestion
    if ((index < MAX_ENDPOINT_COUNT) && (emAfEndpoints[index].endpoint != kInvalidEndpointId) &&
```

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/util/attribute-storage.cpp (hunk 1 above)
Comment (chiragatal) [reply src/app/util/attribute-storage.cpp]: Changed

File: src/app/util/attribute-storage.cpp (hunk 2 above)
Comment (chiragatal) [reply src/app/util/attribute-storage.cpp]: Changed

File: src/app/util/attribute-storage.cpp (hunk 3 above)
Comment (chiragatal) [reply src/app/util/attribute-storage.cpp]: Changed

**Pull request merged**